File: kafka-beginner/kafka-producer-wikimedia/src/main/java/br/com/alanms/demos/wikimedia/WikimediaChangeHandler.java
package br.com.alanms.demos.wikimedia;

import com.launchdarkly.eventsource.MessageEvent;
import com.launchdarkly.eventsource.background.BackgroundEventHandler;
import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class WikimediaChangeHandler implements BackgroundEventHandler {

    private final Logger log = LoggerFactory.getLogger(WikimediaChangeHandler.class.getSimpleName());

    private final KafkaProducer<String, String> kafkaProducer;

    private final String topic;

    public WikimediaChangeHandler(KafkaProducer<String, String> kafkaProducer, String topic) {
        this.kafkaProducer = kafkaProducer;
        this.topic = topic;
    }

    @Override
    public void onOpen() {
        // nothing here
    }

    @Override
    public void onClosed() {
        kafkaProducer.close();
    }

    @Override
    public void onMessage(String s, MessageEvent messageEvent) {
        log.info(messageEvent.getData());
        // asynchronous
        kafkaProducer.send(new ProducerRecord<String, String>(topic, messageEvent.getData()));
    }

    @Override
    public void onComment(String s) throws Exception {
        // nothing here
    }

    @Override
    public void onError(Throwable t) {
        log.error("Error in Stream Reading", t);
    }
}

File: kafka-producer-wikimedia-spring/src/main/kotlin/com/alanms/demos/wikimedia/spring/WikimediaChangeHandler.kt
package com.alanms.demos.wikimedia.spring

import com.launchdarkly.eventsource.MessageEvent
import com.launchdarkly.eventsource.background.BackgroundEventHandler
import org.apache.kafka.clients.producer.ProducerRecord
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate

class WikimediaChangeHandler(private val kafkaTemplate: KafkaTemplate<String, String>): BackgroundEventHandler {

    private val log = LoggerFactory.getLogger(WikimediaChangeHandler::class.simpleName)

    override fun onOpen() {
        // nothing here
    }

    override fun onClosed() {
        // nothing here
    }

    override fun onMessage(s: String, messageEvent: MessageEvent) {
        log.info(messageEvent.data)
        val producerRecord = ProducerRecord<String, String>(TOPIC_NAME, messageEvent.data)
        // asynchronous
        kafkaTemplate.send(producerRecord)
    }

    override fun onComment(c: String?) {
        // nothing here
    }

    override fun onError(t: Throwable?) {
        log.error("Error in Stream Reading", t)
    }
}
